dhcpv6/option_ntp_server.go

```diff
@@ -0,0 +1,157 @@
+
+// Code returns the suboption code.
+func (n *NTPSuboptionMCAddr) Code() NTPSuboptionCode {
+	return NTPSuboptionMCAddrCode
+}
+
+// ToBytes returns the byte serialization of the suboption.
+func (n *NTPSuboptionMCAddr) ToBytes() []byte {
+	buf := uio.NewBigEndianBuffer(nil)
+	buf.Write16(uint16(NTPSuboptionMCAddrCode))
+	buf.Write16(uint16(net.IPv6len))
+	buf.WriteBytes(net.IP(*n).To16())
+	return buf.Data()
+}
+
+func (n *NTPSuboptionMCAddr) String() string {
+	return fmt.Sprintf("Multicast Address: %s", net.IP(*n).String())
+}
+
+// NTPSuboptionSrvFQDN is NTP_SUBOPTION_SRV_FQDN according to RFC 5908.
+type NTPSuboptionSrvFQDN rfc1035label.Labels
+
+// Code returns the suboption code.
+func (n *NTPSuboptionSrvFQDN) Code() NTPSuboptionCode {
+	return NTPSuboptionSrvFQDNCode
+}
+
+// ToBytes returns the byte serialization of the suboption.
+func (n *NTPSuboptionSrvFQDN) ToBytes() []byte {
+	buf := uio.NewBigEndianBuffer(nil)
+	buf.Write16(uint16(NTPSuboptionSrvFQDNCode))
+	l := rfc1035label.Labels(*n)
+	buf.Write16(uint16(l.Length()))
+	buf.WriteBytes(l.ToBytes())
+	return buf.Data()
+}
+
+func (n *NTPSuboptionSrvFQDN) String() string {
+	l := rfc1035label.Labels(*n)
+	return fmt.Sprintf("Server FQDN: %s", l.String())
+}
+
+// NTPSuboptionSrvAddr is the value of NTP_SUBOPTION_SRV_ADDR according to RFC 5908.
+const (
+	NTPSuboptionSrvAddrCode = NTPSuboptionCode(1)
+	NTPSuboptionMCAddrCode  = NTPSuboptionCode(2)
+	NTPSuboptionSrvFQDNCode = NTPSuboptionCode(3)
+)
+
+// NTPSuboptions is a collection of NTPSuboption.
+type NTPSuboptions []NTPSuboption
+
+// FromBytes parses a sequence of bytes into an NTPSuboptions object.
+func (o *NTPSuboptions) FromBytes(data []byte) error {
+	buf := uio.NewBigEndianBuffer(data)
+	for buf.Has(4) {
+		code := NTPSuboptionCode(buf.Read16())
+		length := int(buf.Read16())
+		data, err := buf.ReadN(length)
+		if err != nil {
+			return fmt.Errorf("failed to read %d bytes for suboption: %w", length, err)
+		}
+		switch code {
+		case NTPSuboptionSrvAddrCode, NTPSuboptionMCAddrCode:
+			if length != net.IPv6len {
+				return fmt.Errorf("invalid suboption length, want %d, got %d", net.IPv6len, length)
+			}
+			var so NTPSuboption
+			switch code {
+			case NTPSuboptionSrvAddrCode:
+				sos := NTPSuboptionSrvAddr(data)
+				so = &sos
+			case NTPSuboptionMCAddrCode:
+				som := NTPSuboptionMCAddr(data)
+				so = &som
+			}
+			*o = append(*o, so)
+		case NTPSuboptionSrvFQDNCode:
+			l, err := rfc1035label.FromBytes(data)
+			if err != nil {
+				return fmt.Errorf("failed to parse rfc1035 labels: %w", err)
+			}
+			// TODO according to rfc3315, this label must not be compressed.
+			// Need to add support for compression detection to the
+			// `rfc1035label` package in order to do that.
+			so := NTPSuboptionSrvFQDN(*l)
+			*o = append(*o, &so)
+		}
+	}
+	return buf.FinError()
+}
+
+// OptNTPServer is an option NTP server as defined by RFC 5908.
+type OptNTPServer struct {
+	Suboptions NTPSuboptions
+}
+
```

Comment:
Are you missing a function to parse the `OptNTPServer`?

Also, I think this is a good place where to use the [FromBytesWithParser](https://github.com/insomniacslk/dhcp/blob/d4b282fe8d4cf3f8023e01e0f17cb74c510ad228/dhcpv6/options.go#L192) `Options` method.